### PR TITLE
Decode ABI method added

### LINKED ISSUE #6220: Add method decodeAbi to method interface
The web3 method signature already has the encodeABI() method.
``contract.method.name().encodeABI()``
Im requesting that decodeABI() method be added to the signature
``contract.method.name().decodeABI(result)``
This is usefull when using multicall contracts, since we retrieve result hashes directly from the multicall contract

### PR DESCRIPTION
**Pull request by jerkly**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](https://github.com/web3/web3.js/blob/4.x/CONTRIBUTIONS.md).

Fixes [#6220](https://github.com/web3/web3.js/issues/6220)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-eth-contract/src/contract.ts [hunk 1]
```diff
@@ -1116,6 +1116,7 @@ export class Contract<Abi extends ContractAbi>
 					}),
 
 				encodeABI: () => encodeMethodABI(methodAbi, abiParams),
+				decodeABI: (data?: string) => decodeMethodReturn(methodAbi, data),
```
Comment (Muhammad-Altabba) [inline packages/web3-eth-contract/src/contract.ts:1164]: Thanks for you contribution,
Could you please, mention this added functionality in the related CHANGELOG.md file?
And could you please add a Unit test for this?

**Review by avkos — APPROVED** [review]
LGTM except @Muhammad-Altabba  comment. Thanks for your contribution

File: packages/web3-eth-contract/src/contract.ts (hunk 1 above)
Comment (jdevcs) [reply packages/web3-eth-contract/src/contract.ts:1164]: @jerkly Thanks

**Comment by Muhammad-Altabba** [comment]
Hi @jerkly,
Do you like to finalize this PR? 
Many thanks,

**Comment by Muhammad-Altabba** [comment]
The work for this change will be done at https://github.com/web3/web3.js/pull/6950

**Pull request closed without merging**